pkg/kubelet/server.go

```diff
@@ -117,22 +127,66 @@ type HostInterface interface {
 }
 
 // NewServer initializes and configures a kubelet.Server object to handle HTTP requests.
-func NewServer(host HostInterface, enableDebuggingHandlers bool) Server {
+func NewServer(host HostInterface, auth AuthInterface, enableMetricsHandlers, enableDebuggingHandlers bool) Server {
 	server := Server{
 		host:        host,
+		auth:        auth,
 		restfulCont: restful.NewContainer(),
 	}
+	server.InstallAuthFilter()
 	server.InstallDefaultHandlers()
+	if enableMetricsHandlers {
+		server.InstallMetricsHandlers()
+	}
 	if enableDebuggingHandlers {
 		server.InstallDebuggingHandlers()
 	}
+
 	return server
 }
 
+// InstallAuthFilter installs authentication filters with the restful Container.
+func (s *Server) InstallAuthFilter() {
+	s.restfulCont.Filter(func(req *restful.Request, resp *restful.Response, chain *restful.FilterChain) {
+		// Authenticate
+		u, ok, err := s.auth.AuthenticateRequest(req.Request)
+		if err != nil {
+			resp.WriteError(http.StatusInternalServerError, err)
+			return
+		}
+		if !ok {
+			resp.WriteErrorString(http.StatusUnauthorized, "Unauthorized")
+			return
+		}
+
+		// Get authorization attributes
+		attrs := s.auth.GetRequestAttributes(u, req.Request)
+
+		// Authorize
+		if err := s.auth.Authorize(attrs); err != nil {
+			resp.WriteErrorString(http.StatusForbidden, "Forbidden")
```

Comment:
I'd like to see information closer to the openshift forbidden message.  Knowing who the server thinks you are is valuable, along with knowing how your request parsed.  It seems silly, but it makes sense for more complicated client command interactions that run a series of related requests.  That way you know which permissions to request from the admin.